src/lib/telldus-accessory.js

```diff
@@ -74,12 +74,19 @@ class TelldusAccessory {
     switch(characteristic.props.format) {
       case this.Characteristic.Formats.BOOL:
         (value ? TDtool.on(this.id) : TDtool.off(this.id)).then(out => {
-          callback(null, out.endswith('Success'))
-        }, error => callback(new Error(error)))
-        break
+          return out.indexOf('Success') > -1 ? callback() : Promise.reject(out)
+
+
+          // FIXME: This does not appear to actually be raising an error to
+          //        Homebridge, check out http://goo.gl/RGuILo . Same as below.
+        }, error => { callback(new Error(error)) })
+	break
       case this.Characteristic.Formats.INT:
-        TDtool.setDimLevel(value).then(() => TDtool.dim(this.id)).then(out => {
-          callback(null, out.endswith('Success'))
+        TDtool.dimlevel(value*255.0/100, this.id).then(out => {
```

Comment:
This module actually already contains a method for converting this value, look at https://github.com/amlinger/homebridge-telldus-tdtool/blob/master/src/lib/telldus-accessory.js#L8 and see if that suits your needs :)